modules/fx/arrow-fx/src/main/kotlin/arrow/fx/internal/IOBracket.kt

```diff
@@ -83,7 +83,7 @@ internal object IOBracket {
     IO.Async { conn, cb ->
       Platform.trampoline {
         val frame = EnsureReleaseFrame<A>(release)
-        val onNext = source.flatMap(frame)
+        val onNext = IO.Bind(source, frame)
```

Comment:
Would you be so kind as to hint me on why this makes a difference? Not a full explanation, just something which scent I can track.

Reply:
- `IO.Bind(source, f)` is the ADT case that represents `source.flatMap(source)`.

- `IOFrame : (A) -> IO<A>` is mark functions as error handlers, which can be composed as `source.flatMap(IOFrame.ErrorHandler(...))`

- `flatMap` is optimized for `IO.raiseError` to not execute the `flatMap` since it short-circuits. The missing piece was that it should've scheduled it if it was an `IOFrame` since it cannot skip error handlers.